common/lib/dependabot/file_fetchers/base.rb

```diff
@@ -233,10 +233,19 @@ def _cloned_repo_contents(relative_path)
         Dir.entries(repo_path).map do |name|
           next if [".", ".."].include?(name)
 
+          absolute_path = File.join(repo_path, name)
+          type = if File.symlink?(absolute_path)
+                   "symlink"
+                 elsif Dir.exist?(absolute_path)
+                   "dir"
+                 else
+                   "file"
+                 end
```

Comment:
This private method is indirectly tested through the file fetcher spec, only `file` types are selected here https://github.com/dependabot/dependabot-core/blob/main/terraform/lib/dependabot/terraform/file_fetcher.rb#L62

This `type` mirrors what we get back from github.